implementations/rust/ockam/ockam_api/src/bootstrapped_identities_store.rs

```diff
@@ -87,17 +85,46 @@ impl IdentityAttributeStorageWriter for BootstrapedIdentityStore {
         }
     }
 
+    async fn put_attribute_value(
+        &self,
+        subject: &IdentityIdentifier,
+        attribute_name: &str,
+        attribute_value: &str,
+    ) -> Result<()> {
+        self.repository
+            .put_attribute_value(subject, attribute_name, attribute_value)
+            .await
+    }
+
     async fn delete(&self, identity: &IdentityIdentifier) -> Result<()> {
-        self.storage.delete(identity).await
+        self.repository.delete(identity).await
     }
 }
 
-impl IdentityAttributeStorage for BootstrapedIdentityStore {
-    fn as_identity_attribute_storage_reader(&self) -> Arc<dyn IdentityAttributeStorageReader> {
+#[async_trait]
+impl IdentitiesReader for BootstrapedIdentityStore {
+    async fn get_identity(&self, identifier: &IdentityIdentifier) -> Result<Option<Identity>> {
+        self.repository.get_identity(identifier).await
+    }
+}
+
+#[async_trait]
+impl IdentitiesWriter for BootstrapedIdentityStore {
+    async fn put_identity(&self, identity: &Identity) -> Result<()> {
+        self.repository.put_identity(identity).await
+    }
+
+    async fn update_known_identity(&self, identity: &Identity) -> Result<()> {
```

Comment:
In this particular case, I found confusing the put/update terms. Is the `put` method overriding the identity if already exists?

Reply:
Yes it does. I agree this is not fantastic naming. I exposed `put_identity` because it looked like a more fundamental operation than `update_known_identity` which can be implemented by any client accessing `put_identity`. 
What I can do is hide `put_identity` which is actually not used anywhere in the codebase except for the implementation of `update_known_identity`.

Reply:
Done